### PR TITLE
Videos UI: Select next video for the user on load

### LINKED ISSUE #58257: Videos UI: Queue the user's next video on load.
Without user progress tracking, we've been automatically loading the component with the first video queued up. We now need to open the user's next video on component load.

Requires: #58256

### PR DESCRIPTION
**Pull request by andres-blanco**
This PR selects the corresponding video for the user depending on what she has marked as seen.

The algorithm for selecting the video is:
1. If the user has seen all or none of the videos  => **First video is selected**.
2. Try to select the first video in the sequence that has not been seen by user.

#### Demo with first video marked as viewed

Closes https://github.com/Automattic/wp-calypso/issues/58257

### CODE REVIEW

File: client/components/videos-ui/index.jsx [hunk 1]
```diff
@@ -51,11 +51,23 @@ const VideosUi = ( { headerBar, onBackClick = () => {} } ) => {
...
-			setSelectedVideoIndex( Object.keys( course.videos ).indexOf( initialVideoId ) );
+			setSelectedVideoIndex( videoKeys.indexOf( initialVideoId ) );
+		}
+
+		const viewedVideos = Object.keys( userCourseProgression );
+		if ( viewedVideos.length > 0 ) {
+			const nextVideos = videoKeys.slice(
+				( videoKeys.indexOf( viewedVideos.at( -1 ) ) + 1 ) % videoKeys.length
+			);
+			const nextVideo = nextVideos.find( ( x ) => ! viewedVideos.includes( x ) );
```
Comment (kwight) [inline client/components/videos-ui/index.jsx]: Let's not use meaningless variables, and instead call them what they are; how about `videoSlug`?

File: client/components/videos-ui/index.jsx [hunk 2]
```diff
@@ -51,11 +51,23 @@ const VideosUi = ( { headerBar, onBackClick = () => {} } ) => {
 	} );
 
 	useEffect( () => {
+		const videoKeys = Object.keys( course.videos );
```
Comment (kwight) [inline client/components/videos-ui/index.jsx]: I think you need a null guard here, I wasn't able to open the modal my first try:

Screen Shot 2021-11-25 at 9 04 19 AM

```suggestion
		const videoKeys = Object.keys( course?.videos );
```

**Comment by kwight** [comment]
Hm, that's not the algorithm I expected; I'd thought we would open the first incomplete video in the sequence, regardless of completions. For instance, I've got only the third video complete; I expected to see the first video open on load, not the fourth. (I think it might make for a simpler implementation too.)

Screen Shot 2021-11-25 at 9 07 09 AM

**Comment by andres-blanco** [comment]
> Hm, that's not the algorithm I expected; I'd thought we would open the first incomplete video in the sequence, regardless of completions. For instance, I've got only the third video complete; I expected to see the first video open on load, not the fourth. (I think it might make for a simpler implementation too.)
> 
> Screen Shot 2021-11-25 at 9 07 09 AM

I was thinking that if you skipped a video you probably don't want it to be selected by default on the next load. But you are right in that the implementation is simpler if I just picked the first uncompleted video.

**Comment by kwight** [comment]
> I was thinking that if you skipped a video you probably don't want it to be selected by default on the next load. But you are right in that the implementation is simpler if I just picked the first uncompleted video.

Let's go with the simpler version for now, and @ollierozdarz can chime in if he prefers other behaviour.

File: client/components/videos-ui/index.jsx [hunk 3]
```diff
@@ -50,13 +50,26 @@ const VideosUi = ( { headerBar, footerBar } ) => {
...
 			const initialVideoId = 'find-theme';
 			setCurrentVideoKey( initialVideoId );
-			setSelectedVideoIndex( Object.keys( course.videos ).indexOf( initialVideoId ) );
+			setSelectedVideoIndex( videoSlugs.indexOf( initialVideoId ) );
 		}
-	}, [ currentVideoKey, course ] );
+
+		const viewedSlugs = Object.keys( userCourseProgression );
+		if ( viewedSlugs.length > 0 ) {
+			const nextSlug = videoSlugs.find( ( x ) => ! viewedSlugs.includes( x ) );
```
Comment (kwight) [inline client/components/videos-ui/index.jsx]: I think you misunderstood me here; it's the `x` variable I didn't like. :)

**Comment by kwight** [comment]
Lovely!

Screen Shot 2021-11-25 at 11 23 40 AM

**Review by kwight — APPROVED** [review]
Just fix the `x` variable and 🚢

File: client/components/videos-ui/index.jsx (hunk 3 above)
Comment (andres-blanco) [reply client/components/videos-ui/index.jsx]: 🤦🏻‍♂️, I'll fix that

**Pull request merged**